eb_hooks.py

```diff
@@ -3,21 +3,82 @@
 import os
 
 from easybuild.tools.build_log import EasyBuildError, print_msg
+from easybuild.tools.config import build_option, update_build_option
 from easybuild.tools.systemtools import POWER, get_cpu_architecture
 
 
+def get_eessi_envvar(eessi_envvar):
+    """Get an EESSI environment variable from the environment"""
+
+    eessi_envvar_value = os.getenv(eessi_envvar)
+    if eessi_envvar_value is None:
+        raise EasyBuildError("$%s is not defined!", eessi_envvar)
+
+    return eessi_envvar_value
+
+
+def get_rpath_override_dir(software_name):
+    # determine path to installations in software layer via $EESSI_SOFTWARE_PATH
+    eessi_software_path = get_eessi_envvar('EESSI_SOFTWARE_PATH')
```

Comment:
`$EESSI_SOFTWARE_PATH` is not set currently in the build environment.

It is set by the `init` script, but if the hooks are going to rely on it, then we should make sure it's defined there too...

The build script sets `$EESSI_PREFIX`, should we be using that instead here?
Or perhaps use `$CVMFS_REPO`:

```python
cvmfs_repo = get_eessi_envvar('CVMFS_REPO')
os.path.join(cvmfs_repo, 'host_injections', eessi_pilot_version)
```

It probably wouldn't hurt to *always* grab these environment variables via a startup hook, to ensure problems like this are caught early on in the build process of the software layer?

Reply:
But the content of `EESSI_SOFTWARE_PATH` is quite different and I actually want this full path (it would allow you to maintain a single location for all archs):
```
[ocaisa1@node1 ~]$ echo $EESSI_SOFTWARE_PATH/
/cvmfs/pilot.eessi-hpc.org/2021.03/software/linux/x86_64/amd/zen2/
[ocaisa1@node1 ~]$ echo $EESSI_PREFIX/
/cvmfs/pilot.eessi-hpc.org/2021.03/
```

Reply:
The value is available as `EASYBUILD_INSTALLPATH`, should I use that instead.

I understand that it is debatable whether all those subdirs are so useful for this use case, a convincing argument may sway me.

Reply:
I think it's wiser to use `$EESSI_*` environment variables, rather than making assumptions about the EasyBuild `installpath` value... Although perhaps a check whether the value of the `$EESSI_*` environment variables and the active EasyBuild `installpath` wouldn't hurt, to avoid surprises? Not sure how strict we want to make this...

Can you clarify (in a comment in the hook) via an example what the full path is you want to inject?

Is that something like `/cvmfs/pilot.eessi-hpc.org/host_injections/2021.03/software/linux/x86_64/amd/zen2/rpath_overrides/GROMACS` ?

I'm not sure how that would allow you to maintain a single location for all archs? Through symlinks?

Reply:
The location would be
```
/cvmfs/pilot.eessi-hpc.org/host_injections/2021.03/software/linux/x86_64/generic/rpath_overrides/OpenMPI
```
Since we are overriding where MPI is to be found (same location for many packages).

The script that actually creates this structure (and is still to be written) could indeed default to using symlinks with the assumption that what you want to do is universal.

Reply:
For now, I think we can let the build script also define `$EESSI_SOFTWARE_PATH` to be equal to `$EASYBUILD_INSTALLPATH`, so we can get this PR merged.

Longer term, it makes sense to have a single script that sets up the EESSI environment that is used by both the `init` script and the build script, so ensure consistency.

Reply:
see https://github.com/ocaisa/software-layer/pull/2